### ISSUE TITLE
📎 Port `function-component-definition` rule from eslint react plugin

### ISSUE THREAD

**Issue by gameroman** [biomejs/biome#10320]
### Description

❌ Incorrect:
```tsx
const MyComponent = (props) => {
  return <div>{props.name}</div>;
};
```

✅ Correct:
```tsx
function MyComponent(props) {
  return <div>{props.name}</div>;
}
```

https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/function-component-definition.md

**Comment by GrantTotinov** [comment]
I’d like to work on this issue if no one else is currently implementing it.

**Comment by Dotify71** [comment]
Hi @gameroman! I would like to work on this issue. I'll be porting the rule from the ESLint React plugin and ensuring support for all component definition styles. Thanks!
